_rules/text-node-contrast-afw4f7.md

```diff
@@ -0,0 +1,337 @@
+	Some text in English
+</p>
+```
+
+#### Passed Example 5
+
+18 point text with a 3:1 contrast.
+
+```html
+<p style="color: #000; font-size:18pt; background: #666;">
+	Some text in English
+</p>
+```
+
+#### Passed Example 6
+
+14 point bold text with a 3:1 contrast.
+
+```html
+<p style="color: #000; font-size:14pt; font-weight:700; background: #666;">
+	Some text in English
+</p>
+```
+
+#### Passed Example 7
+
+Decorative text with insufficient contrast.
+
+```html
+<p>Helvetica is a widely used sans-serif typeface developed in 1957 by Max Miedinger and Eduard Hoffmann.</p>
+<p style="font-family: helvetica; background: #EEE; color: #777;" aria-hidden="true">
+	The quick brown fox jumps over the lazy dog.
+</p>
+```
+
+#### Passed Example 8
+
+Text that does not express anything in human language.
+
+```html
+<p style="color: #000; background: #666;">
+	----=====++++++++___________***********%%%%%%%%%%%±±±±@@@@@@@@
+</p>
+```
+
+#### Passed Example 9
+
+Text that is in a shadow DOM tree with sufficient contrast.
+
+```html
+<p style="color: #333; background: #fff;" id="p"></p>
+<script>
+  const shadowRoot = document.getElementById('p').attachShadow({ mode: 'open' })
+  shadowRoot.textContent = 'Some text in English'
+</script>
+```
+
+### Failed
+
+#### Failed Example 1
+
+Text with insufficient contrast on a plain background.
+
+```html
+<p style="color: #AAA; background: white;">
+	Some text in English
+</p>
+```
+
+#### Failed Example 2
+
+Text with insufficient contrast on a gradient background.
+
+```html
+<p style="color: #AAA; background: linear-gradient(to right, #FFF, #00F);">
+	Some text in English
+</p>
+```
+
+#### Failed Example 3
+
+Text with insufficient contrast on a background image.
+
+```html
+<p
+	style="color: #555; height:50px; padding-top:25px; background: black no-repeat -20px -20px url('../test-assets/black-hole.jpeg');"
+>
+	Black hole sun
+</p>
+```
+
+#### Failed Example 4
+
+Text that fails because of alpha transparancy.
+
+```html
+<p style="color: rgba(0,0,0,.3); background: #FFF">
+	Some text in English
+</p>
+```
+
+#### Failed Example 5
+
+Text that fails because of CSS transparent.
+
+```html
+<div style="background: #FFF">
+	<p style="color: #000; opacity: .3">
+		Some text in English
+	</p>
+</div>
+```
+
+#### Failed Example 6
+
+Text that is in a shadow DOM tree with insufficient contrast.
+
+```html
+<p style="color: #aaa; background: #fff;" id="p"></p>
+<script>
+  const shadowRoot = document.getElementById('p').attachShadow({ mode: 'open' })
+  shadowRoot.textContent = 'Some text in English'
+</script>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+Invisible text.
+
+```html
+<p style="display: none">Some invisible text in English</p>
+```
+
+#### Inapplicable Example 2
+
+Text with the same foreground and background colors.
+
+```html
+<p style="color: white; background: white;">Some white on white text in English</p>
+```
+
```

Comment:
Should we add an example where text is invisible because of positioning (the usual "invisible but in accessibility tree" example)?